grpc-dump/dump/dump_interceptor.go

```diff
@@ -5,17 +5,43 @@ import (
 	"fmt"
 	"io"
 	"strings"
+	"sync"
 
 	"github.com/bradleyjkemp/grpc-tools/internal"
 	"github.com/bradleyjkemp/grpc-tools/internal/proto_decoder"
-
+	"github.com/bradleyjkemp/grpc-tools/internal/proto_descriptor"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/dynamic"
 	"github.com/sirupsen/logrus"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
 	"google.golang.org/grpc/status"
 )
 
 // dump interceptor implements a gRPC.StreamingServerInterceptor that dumps all RPC details
+var mu sync.Mutex
+
+type pbm struct {
+	*dynamic.Message
+}
+
+func (p *pbm) MarshalJSON() ([]byte, error) {
+	fd := make([]*desc.FileDescriptor, 0)
+	proto_descriptor.MsgDesc.Lock()
```

Comment:
As `proto_descriptor.MsgDesc` is only going to be mutated during startup (i.e. when loading proto files), I wonder if this step of converting to creating a new dynamic.AnyResolver needs to be done every time?

Should (/could) this just be done once during proto file loading?

Reply:
Good point. Beside that, and as I mentioned in #96,
> is probably not clean in the general way grpc-tools is designed

`proto_descriptor.MsgDesc` is a global varible and I'm not sure if this is the right way to bring these message describtors into the dump interceptor. It was the fastest way to have them be available, but should be probably done in a cleaner way. Wdyt?